antarest/study/storage/patch_service.py

```diff
@@ -1,28 +1,48 @@
 from pathlib import Path
-from typing import Union
+from typing import Union, Optional
+
+from pydantic import ValidationError
 
-from antarest.core.model import JSON
 from antarest.study.model import Patch, PatchOutputs, RawStudy
+from antarest.study.repository import StudyMetadataRepository
 from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
 from antarest.study.storage.variantstudy.model.dbmodel import (
     VariantStudy,
 )
 
 
 class PatchService:
-    def get(self, study: Union[RawStudy, VariantStudy, FileStudy]) -> Patch:
-        patch_path = (
-            Path(study.path)
-            if not isinstance(study, FileStudy)
-            else study.config.study_path
-        ) / "patch.json"
+    def __init__(self, repository: Optional[StudyMetadataRepository] = None):
+        self.repository = repository
+
+    def get(
+        self, study: Union[RawStudy, VariantStudy], get_from_file: bool = False
+    ) -> Patch:
+        patch = Patch()
+
+        if not get_from_file:
+            try:
+                patch = Patch.parse_raw(study.additional_data.patch)
+            except (AttributeError, ValidationError):
+                get_from_file = True
+
+        if get_from_file:
```

Comment:
no need of "if" here

Reply:
It's usually easier to read if there is only one return, this is the reason of this if, and not the early return

Reply:
well i don't know, it is best practice to return earlier.